lib/src/shared/presentation/widgets/language_switch_button.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:interns2025b_mobile/src/shared/presentation/providers/localization_controller_provider.dart';
+
+class LanguageSwitchButton extends ConsumerWidget {
+  const LanguageSwitchButton({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final controller = ref.watch(localizationControllerProvider);
+    final notifier = ref.read(localizationControllerProvider);
+
+    final currentLang = controller.locale.languageCode;
+    final isPolish = currentLang == 'pl';
+    final label = isPolish ? 'PL' : 'EN';
+
+    return OutlinedButton.icon(
+      onPressed: () {
+        final newLang = isPolish ? 'en' : 'pl';
+        notifier.setLocale(newLang);
+      },
```

Comment:
You're referencing the same `localizationControllerProvider` in both `watch` and `read` calls. Wouldn't using `watch` alone be sufficient?


```suggestion
    final controller = ref.watch(localizationControllerProvider);
   
    final currentLang = controller.locale.languageCode;
    final isPolish = currentLang == 'pl';
    final label = isPolish ? 'PL' : 'EN';

    return OutlinedButton.icon(
      onPressed: () {
        final newLang = isPolish ? 'en' : 'pl';
        controller.setLocale(newLang);
      },
```